### PR TITLE
GDB-8085: implement update query flow

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
Separates execution query flow depends on the query mode:
- "query" mode;
- "update" mode.

## Why
Those two mode has some differences. All queries of "query" mode have total result elements. Update queries updates the repository statements and as result the affected repository can change its statements. As result the user executed the query can add or remove statements.

- "query" mode flow:
- execute query;
- shows results;
- execute count query;
- shows results info message.
- "update" mode flow:
- call the function "beforeUpdateQuery". This function can be passed from "ontotext-yasgui-web-component" client. The component client can check some precondition and can stop execution of the query.
- call the function "getRepositoryStatementsCount". This function have to be passed from  "ontotext-yasgui-web-component" client if client want message result info about added, removed statements to be shown;
- execute the query;
- call the function "getRepositoryStatementsCount" again;
- calculates count of affected statements;
- shows message result info to the user.

## How

The "query" mode is already implemented. Added implementation to update flow as described above. Added configurations for "beforeUpdateQuery" and "getRepositoryStatementsCount" functions.

### CODE REVIEW

File: Yasgui/packages/yasgui/src/Tab.ts [hunk 1]
```diff
@@ -514,6 +516,17 @@ export class Tab extends EventEmitter {
     }
   };
 
+  handleCountAffectedRepositoryStatementsChanged = (_yasqe: Yasqe, countAffectedRepositoryStatements: number) => {
+    if (this.yasr?.results) {
+      const response = this.persistentJson.yasr.response || {};
```
Comment (svilenvelikov) [inline Yasgui/packages/yasgui/src/Tab.ts:521]: I think you don't need this default `{}` here. Just check for the response.

File: Yasgui/packages/yasqe/src/sparql.ts [hunk 2]
```diff
@@ -50,6 +50,51 @@ export function getAjaxConfig(
...
+    yasqe.config.getRepositoryStatementsCount().then((repositoryStatementsCount) => {
+      if (initialRepositoryStatementsCount !== undefined && repositoryStatementsCount !== undefined) {
+        const affectedStatementsCount = repositoryStatementsCount - initialRepositoryStatementsCount;
+        yasqe.emit("countAffectedRepositoryStatementsChanged", affectedStatementsCount);
+        yasqe.emit("countAffectedRepositoryStatementsPersisted");
+      }
+    });
+  };
+
+  return await checkQueryPreconditions()
```
Comment (svilenvelikov) [inline Yasgui/packages/yasqe/src/sparql.ts:86]: It's quite confusing when promises and async-await approaches are intermixed. The implementation in yasgui primarily relies on async-await so you can stick with it or just use promises and don't mix them up.

File: Yasgui/packages/yasqe/src/sparql.ts [hunk 3]
```diff
@@ -74,7 +119,7 @@ export async function executeQuery(yasqe: Yasqe, config?: YasqeAjaxConfig): Prom
     return await req.then(
       (result) => {
         let hasMorePage = false;
-        if (yasqe.config.paginationOn) {
+        if ("update" !== yasqe.getQueryMode() && yasqe.config.paginationOn) {
```
Comment (svilenvelikov) [inline Yasgui/packages/yasqe/src/sparql.ts:122]: You can use the `isUpdateQuery` method from yasqe I believe.

File: cypress/e2e/editor-actions/exequte-update-query.spec.cy.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import {YasqeSteps} from '../../steps/yasqe-steps';
+import {YasrTablePluginSteps} from '../../steps/yasr-table-plugin-steps';
+import {ExecuteUpdateQueryPageSteps} from '../../steps/pages/execute-update-query-page-steps';
+
+describe('Execution of update query', () => {
```
Comment (svilenvelikov) [inline cypress/e2e/editor-actions/exequte-update-query.spec.cy.ts:5]: Execution -> Execute

Comment (svilenvelikov) [reply cypress/e2e/editor-actions/exequte-update-query.spec.cy.ts:5]: I think that the file is badly formatted.

File: Yasgui/packages/yasqe/src/sparql.ts (hunk 3 above)
Comment (boyan-tonchev) [reply Yasgui/packages/yasqe/src/sparql.ts:122]: Hm. I was use it :) . Maybe I made mistake in some rebeising.

**Comment by boyan-tonchev** [comment]
Jenkins retry

**Comment by boyan-tonchev** [comment]
Jenkins retry

**Review by svilenvelikov — APPROVED** [review]

**Pull request merged**